src/clib/pio_nc.c

```diff
@@ -2987,7 +2987,7 @@ int PIOc_def_dim(int ncid, const char *name, PIO_Offset len, int *idp)
         }
 
         assert(file->num_dim_vars < PIO_MAX_DIMS);
-        file->dim_names[file->num_dim_vars] = strdup(name);
+        file->dim_names[file->num_dim_vars] = pio_strdup(name);
```

Comment:
Can you also handle pio_strdup() fails (NULLs returned by the function) too?

Reply:
Good point, but I think it can also be done in a separate PR (e.g. part of HDF5/ADIOS code cleanup/refactoring).